server/src/main/java/org/opensearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -223,6 +255,10 @@ protected ClusterStatsNodeResponse nodeOperation(ClusterStatsNodeRequest nodeReq
         );
     }
 
+    private boolean isMetricRequired(boolean applyMetricFilter, Metric metric, Set<Metric> requestedMetrics) {
+        return !applyMetricFilter || requestedMetrics.contains(metric);
```

Comment:
should this be `applyMetricFilter && requestedMetrics.contains(metric)`

Reply:
We want to compute all metrics when the filter is not enabled hence using the OR boolean condition.